features/checkout/paying_for_order/cancelling_payment_request_when_payment_method_is_changed.feature

```diff
@@ -0,0 +1,21 @@
+@paying_for_order
+Feature: Cancelling payment request when payment method is changed
+    In order to pay with the correct payment method
+    As a Customer
+    I want my payment request with wrong payment method to be cancelled
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "Iron Maiden T-Shirt"
+        And the store ships everywhere for Free
+        And the store allows paying Offline
+        And the store has a payment method "PayPal" with a code "PAYPAL" and Paypal Express Checkout gateway
+        And there is a customer "sylius@example.com" that placed an order "#00000022"
+
+    @no-ui @api
+    Scenario: Cancelling the payment request when payment method is changed
+        Given I added product "Iron Maiden T-Shirt" to the cart
+        And I have proceeded selecting "PayPal" payment method
+        When I confirm my order with paypal payment
+        And I change payment method to "Offline" after checkout
+        Then the payment request for payment method "PayPal" should be cancelled
```

Comment:
```suggestion
    @no-ui @api
    Scenario: Cancelling the payment request when payment method is changed
        Given I added product "Iron Maiden T-Shirt" to the cart
        And I have proceeded selecting "PayPal" payment method
        And I confirm my order with paypal payment # <- I'm wondering whether it's necessary as we have "I have proceeded"
        When I change payment method to "Offline" after checkout
        Then the payment request for payment method "PayPal" should be cancelled
```

IMO confirm is a part of preparation, and `When` starts for this scenario when I want to change the payment method. Also I'm wondering whether the `I confirm` can be skipped here.

Reply:
I renamed it a bit, as yeah it might be misleading